### ISSUE TITLE
If a draggable element entity parent is moving while the element is dragged, it will move in local Z

### ISSUE THREAD

**Issue by yaustar** [playcanvas/engine#3748]
Example project: https://playcanvas.com/editor/scene/1282724

Expected behaviour is for the element entity to stay in its current local Z position or the local Z position from when it was started to be dragged.

This issue affects scroll views/elements.

https://user-images.githubusercontent.com/16639049/143869233-a51c7974-fb4e-4563-b80c-5bd2311ef550.mp4

As a workaround, a patch can be made to the ElementDragHelper:

```
(function () {
    const OPPOSITE_AXIS = {
        x: 'y',
        y: 'x'
    };
    pc.ElementDragHelper.prototype._onMove = function (event) {
        if (this._element && this._isDragging && this.enabled && this._element.enabled && this._element.entity.enabled) {
            const currentMousePosition = this._screenToLocal(event);

            if (this._dragStartMousePosition && currentMousePosition) {
                this._deltaMousePosition.copy(currentMousePosition).sub(this._dragStartMousePosition);
                this._deltaHandlePosition.copy(this._dragStartHandlePosition).add(this._deltaMousePosition);
                const currentPosition = this._element.entity.getLocalPosition();

                if (this._axis) {
                    const constrainedAxis = OPPOSITE_AXIS[this._axis];
                    this._deltaHandlePosition[constrainedAxis] = currentPosition[constrainedAxis];
                }

                // Enforce no movement on the Z axis
                this._deltaHandlePosition.z = currentPosition.z;
                this._element.entity.setLocalPosition(this._deltaHandlePosition);
                this.fire('drag:move', this._deltaHandlePosition);
            }
        }
    };
})();
```

**Comment by kungfooman** [comment]
Just tested and this issue is also fixed: https://launch.playcanvas.com/1282724?debug=true&use_local_engine=https://code.playcanvas.com/playcanvas-1.57.0.js

**Comment by yaustar** [comment]
Fixed with #4666
